pyccel/parser/semantic.py

```diff
@@ -2648,6 +2649,19 @@ def _visit_IfSection(self, expr, **settings):
 
     def _visit_If(self, expr, **settings):
         args = [self._visit(i, **settings) for i in expr.blocks]
+        allocations = [arg.get_attribute_nodes(Allocate) for arg in args]
+        var_shapes = [dict() for _ in range(len(args))]
+        for i,allocs in enumerate(allocations):
+            for a in allocs:
+                var_shapes[i][a.variable] = a.shape
```

Comment:
Could this code be a bit faster and cleaner?
```python
var_shapes = [{a.variable : a.shape for a in allocs} for allocs in allocations]

Reply:
I'm not sure. Do you know how this reacts if the same key appears twice?

Reply:
It seems to react the same way as it does in a loop so this is much cleaner :). Fixed in ada1f0e1